### PR TITLE
Implement helper classes and utility functions for query pushdown

### PR DESCRIPTION
**Pull request by VinayLondhe14**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java [hunk 1]
```diff
@@ -54,4 +55,28 @@ public InvalidSchemaException(String message, Throwable t) {
       super(BIGQUERY_INVALID_SCHEMA, message, t);
     }
   }
+
+  /** Exception if query pushdown to BigQuery fails */
+  public static class PushdownException extends BigQueryConnectorException {
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java:60]: Please move exceptions to be top level classes

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQuerySQLStatement.scala [hunk 2]
```diff
@@ -0,0 +1,151 @@
+package com.google.cloud.spark.bigquery.pushdowns
```
Comment (davidrabinowitz) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQuerySQLStatement.scala:1]: Missing license header

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala [hunk 3]
```diff
@@ -1,7 +1,11 @@
 package com.google.cloud.spark.bigquery.pushdowns
```
Comment (davidrabinowitz) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala:1]: Missing license header

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java (hunk 1 above)
Comment (VinayLondhe14) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java]: done

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala [hunk 4]
```diff
@@ -15,4 +35,80 @@ object SparkBigQueryPushdownUtil {
...
+  def blockStatement(stmt: BigQuerySQLStatement): BigQuerySQLStatement =
+    ConstantString("(") + stmt + ")"
+
+  /**
+   * Creates a new BigQuerySQLStatement by taking a list of BigQuerySQLStatement and folding it into one BigQuerySQLStatement separated by a delimiter
+   * @param seq
+   * @param delimiter
+   * @return
+   */
+  def mkStatement(seq: Seq[BigQuerySQLStatement], delimiter: String): BigQuerySQLStatement =
```
Comment (davidrabinowitz) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala:57]: Rename to `makeStatement`

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala [hunk 5]
```diff
@@ -15,4 +35,80 @@ object SparkBigQueryPushdownUtil {
...
+    mkStatement(seq, ConstantString(delimiter).toStatement)
+
+  def mkStatement(seq: Seq[BigQuerySQLStatement], delimiter: BigQuerySQLStatement): BigQuerySQLStatement =
```
Comment (davidrabinowitz) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkBigQueryPushdownUtil.scala:60]: Rename to `makeStatement`

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please add unit test for SparkBigQueryPushdownUtil

**Comment by VinayLondhe14** [comment]
added unit test for SparkBigQueryPushdownUtil

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**